extension/src/project/readOnlyDocument.ts

```diff
@@ -127,20 +127,14 @@ export class ReadonlyDocument implements vscode.TextDocument {
             if (content) {
                 this.initialize(content, this.languageId);
             }
-            let parser = new LispParser(this);
-            try {
-                parser.tokenizeString(this.fileContent, 0);        
-                this._atomsForest = [...parser.atomsForest];    
-            } finally {
-                parser = undefined;
-            }
+            this._docExpression = LispParser.getDocumentSexpression(this.fileContent);  
```

Comment:
Please note that the ReadOnlyDocument is used for project Open&Save.
I suggest that we limit the use of the new parser to new features for now, and gradually use it to replace the old one after well tested.

Reply:
The `_docExpression` is only currently used internally by the ReadOnlyDocument's `get atomsForest()` property. So, both have only ever been used to power new features. I agree on the testing, I wouldn't be the least bit surprised if malformed syntax will cause some some edge case anomalies. I don't really expect those to surface in clear ways until AutoCompletion starts attempting to use this.

Sexpression differences are still the one thing I am most concerned about. Should we add a property to the Sexpression class that can identify whether it is part of the analytical model? We wouldn't have to update any other code, simply initialize the value and this new stuff could explicitly set it to something unique. I could then run through all the "newer" Sexpression functions putting their current work in the "new" code path and leave ToDo notes on the else. At least then it would create some clarity on where that divide happened and what was built for which task.

Reply:
I see, you're right.

Reply:
Another thought on the Sexpressions delima:
Maybe we should create a class that inherits Sexpression or that Sexpression inherits to completely divide what is part of the original formatting context and what is going to be used for responsive analytical work? If you think about it, the Sexpression currently has a ton of formatting specific behaviors that have nothing to do with the core of what an Sexpression really is; which is a container....

Reply:
Yes, it makes great sense.